modules/light-clients/07-tendermint/types/update_test.go

```diff
@@ -279,3 +280,66 @@ func (suite *TendermintTestSuite) TestCheckHeaderAndUpdateState() {
 		}
 	}
 }
+
+func (suite *TendermintTestSuite) TestPruneConsensusState() {
+	ctx := suite.chainA.GetContext().WithBlockTime(suite.now)
+
+	// create 2 consensus states with the clientTime along with the metadata that would be stored by UpdateClient
+	consensusState := types.NewConsensusState(suite.clientTime, commitmenttypes.NewMerkleRoot(suite.header.Header.GetAppHash()), suite.valsHash)
+	suite.chainA.App.IBCKeeper.ClientKeeper.SetClientConsensusState(ctx, clientID, height, consensusState)
+	types.SetProcessedTime(suite.chainA.App.IBCKeeper.ClientKeeper.ClientStore(ctx, clientID), height, uint64(ctx.BlockTime().UnixNano()))
+	types.SetIterationKey(suite.chainA.App.IBCKeeper.ClientKeeper.ClientStore(ctx, clientID), height)
+	nextHeight := height.Increment()
+	suite.chainA.App.IBCKeeper.ClientKeeper.SetClientConsensusState(ctx, clientID, nextHeight, consensusState)
+	types.SetProcessedTime(suite.chainA.App.IBCKeeper.ClientKeeper.ClientStore(ctx, clientID), nextHeight, uint64(ctx.BlockTime().UnixNano()))
+	types.SetIterationKey(suite.chainA.App.IBCKeeper.ClientKeeper.ClientStore(ctx, clientID), nextHeight)
```

Comment:
when you update to the new testing package, I think you can just create and update the client

Reply:
I also don't think you should be manually setting the keys. We should test that regular client creation/updates set these properly

Reply:
Good call, realized I forgot to SetIterationKey in UpdateClient because I was manually setting in tests. 😬